### ISSUE TITLE
IPv6 SDP problems

### ISSUE THREAD

**Issue by dpocock** [jitsi/jitsi#170]
I have two machines both running dual-stack IPv4 and IPv6

I make a SIP call from the softphone on one machine to Jitsi on the other machine

In the SDP offer from the caller, the "c=" line contains "IP6" and an IPv6 address

In the SDP answer from Jitsi, the "c=" line contains "IP4" and an IPv4 address.

Jitsi then starts sending media to the caller using an IPv4-Mapped IPv6 source address [::ffff:192.168.1.128]

The caller could potentially send to Jitsi using one of its IPv4 addresses, but if the caller was a pure-IPv6 host then it wouldn't work.  Why doesn't Jitsi just use IPv6 in the SDP answer?

**Comment by dpocock** [comment]
discussed on users list http://lists.jitsi.org/pipermail/users/2015-October/010290.html

**Comment by 459below** [comment]
I no expert in ICE, NAT, STUN and so forth. But maybe this is related https://github.com/jitsi/jitsi-android/issues/36

**Comment by dpocock** [comment]
It is not related to ICE, NAT or STUN, both machines are on the same LAN

Jitsi doesn't even support ICE or STUN on SIP, they are only supported with XMPP

**Comment by ibauersachs** [comment]
@dpocock What was your test setup for this? Was a SIP proxy/registrar involved? Any DNS resolution?

I tried a call which had IP6 in both the invite and the answer. - two registrarless Jitsi instances on the same network
- both PCs had a DHCP assigned IPv4 address
- both PCs had a statically configured IPv6 address (from a unique local range)
- call to the IPv6-address of the other machine (no DNS involved)

**Comment by dpocock** [comment]
I think I was testing through a repro SIP proxy

When you made the call, was there IPv6 in the offer SDP?  If not, try initiating the call from a host without any IPv4 enabled.

**Comment by ibauersachs** [comment]
Hm, a proxy makes this more difficult to test.

Yes, offer and answer both contained `c= ... IP6 ...`. Not much of a surprise given that it was a call to an IPv6 address and not some DNS name.

**Comment by dpocock** [comment]
Could you test it through the proxy used for the GSoC projects?  test.sip5060.net

**Comment by ibauersachs** [comment]
No, as I still don't have IPv6 on my internet connection :(

**Comment by dpocock** [comment]
Can you try an IPv6 tunnel service?

**Comment by ibauersachs** [comment]
My router seems to block protocol 41 (6to4). Do you have a config for whatever you run on sip5060 that you could share so I can easily setup a local server?

**Comment by ibauersachs** [comment]
pfSense to the rescue and fiddling with the crappy Cablecom router. Will continue investigating...

**Comment by ibauersachs** [comment]
@dpocock Thanks for the config. I had managed to get a tunnel up and used your ws5060 for a while. The c= has always been constructed with what Jitsi used to connect to the proxy. I understand this is problematic (i.e. a (dual-stack )host connecting to the proxy over IPv6 and an IPv4 host cannot 'talk' to each other), but I don't see a real possibility to solve this without ICE.
If the proxy would relay media (and change the sdp for that) it should work. And this is the only supported scenario in Jitsi.

While investigating the issue I fixed several other (IPv6) bugs.
